sklearn/utils/tests/test_validation.py

```diff
@@ -1053,3 +1056,31 @@ def __init__(self, a=1, b=1, *, c=1, d=1):
     with pytest.warns(FutureWarning,
                       match=r"Pass c=3, d=4 as keyword args"):
         A2(1, 2, 3, 4)
+
+
+def test_check_fit_params():
+    X = np.random.randn(4, 2)
+    fit_params = {
+        'list': [1, 2, 3, 4],
+        'tuple': (1, 2, 3, 4),
+        'array': np.array([1, 2, 3, 4]),
+        'sparse-col': sp.csc_matrix([1, 2, 3, 4]).T,
+        'sparse-row': sp.csc_matrix([1, 2, 3, 4]),
+        'scalar-func': accuracy_score,
+        'scalar-class': KNeighborsClassifier,
+        'scalar-int': 1,
+        'scalar-str': 'xxx',
+        'None': None,
+    }
+    result = _check_fit_params(X, fit_params)
+
+    assert isinstance(result['list'], list)
```

Comment:
If you're trying to confirm that it's unchanged, why not use `is` to compare?

Reply:
Since I added the `indices` parameters, I don't need the type checking anymore. With indexing the `is` will not be true anymore.